includes/object-cache.php

```diff
@@ -959,46 +966,9 @@ protected function connect_using_credis( $parameters ) {
             $this->redis = $sentinel->getCluster( WP_REDIS_SENTINEL );
             $args['servers'] = WP_REDIS_SERVERS;
         } elseif ( defined( 'WP_REDIS_CLUSTER' ) || defined( 'WP_REDIS_SERVERS' ) ) {
-            $parameters['db'] = $parameters['database'];
-
-            $is_cluster = defined( 'WP_REDIS_CLUSTER' );
-            $clients = $is_cluster ? WP_REDIS_CLUSTER : WP_REDIS_SERVERS;
-
-            foreach ( $clients as $index => $connection_string ) {
-                // phpcs:ignore WordPress.WP.AlternativeFunctions.parse_url_parse_url
-                $url_components = parse_url( $connection_string );
-
-                parse_str( $url_components['query'], $add_params );
-
-                if ( ! $is_cluster && isset( $add_params['alias'] ) ) {
-                    $add_params['master'] = 'master' === $add_params['alias'];
-                }
-
-                $add_params['host'] = $url_components['host'];
-
-                if ( ! isset( $add_params['alias'] ) ) {
-                    $add_params['alias'] = "redis-$index";
-                }
-
-                $clients[ $index ] = array_merge( $parameters, $add_params );
-            }
-
-            $this->redis = new Credis_Cluster( $clients );
-
-            foreach ( $clients as $index => $_client ) {
-                $connection_string = "{$_client['scheme']}://{$_client['host']}:{$_client['port']}";
-                unset( $_client['scheme'], $_client['host'], $_client['port'] );
-
-                $params = array_filter( $_client );
-
-                if ( $params ) {
-                    $connection_string .= '?' . http_build_query( $params, null, '&' );
-                }
-
-                $clients[ $index ] = $connection_string;
-            }
-
-            $args['servers'] = $clients;
+            throw new Exception(
+                'Clusters not fully supported by bundled Credis library. Please review your Redis Cache configuration.'
+            );
```

Comment:
Does throwing an exception here turn into a fatal error, or a log message?

Reply:
Mostly fatal error that will be displayed in the diagnostics, I used the same approach as the one used when checking for `WP_REDIS_SHARDS`.

![image](https://user-images.githubusercontent.com/27624/178136433-d35c6357-0078-412e-a5a4-763838674c24.png)


But, as we discusses since the library is [no longer actively developed](https://github.com/colinmollenhour/credis/issues/168#issuecomment-1178308804), we will drop the support for `Credis`.